indefinitepagerindicator/src/main/java/com/rbrooks/indefinitepagerindicator/IndefinitePagerIndicator.kt

```diff
@@ -404,8 +450,14 @@ class IndefinitePagerIndicator @JvmOverloads constructor(
             }
         }
 
-        private fun setIntermediateSelectedItemPosition(mostVisibleChild: View?) {
-            with(recyclerView?.findContainingViewHolder(mostVisibleChild)?.adapterPosition!!) {
+        private fun setIntermediateSelectedItemPosition(mostVisibleChild: View) {
+            with(recyclerView?.findContainingViewHolder(mostVisibleChild)?.adapterPosition) {
```

Comment:
we could remove the *with* clause and use let instead. Because we are just using **this** everywhere that way we could also do nothing if not adapter position exists. What do you think?